aspnetcore/client-side/dotnet-interop/index.md

```diff
@@ -0,0 +1,910 @@
+
+The following conditions apply to type mapping and marshalled values:
+
+* The `Array of` column indicates if the .NET type can be marshalled as a JS [`Array`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array). Example: C# `int[]` (`Int32`) mapped to JS `Array` of `Number`s.
+* When passing a JS value to C# with a value of the wrong type, the framework throws an exception in most cases. The framework doesn't perform compile-time type checking in JS.
+* `JSObject`, `Exception`, `Task` and `ArraySegment` create `GCHandle` and a proxy. You can trigger disposal in developer code or allow [.NET garbage collection (GC)](/dotnet/standard/garbage-collection/) to dispose of the objects later. These types carry significant performance overhead.
+* `Array`: Marshaling an array creates a copy of the array in JS or .NET.
+* `MemoryView`
+  * `MemoryView` is a JS class for the .NET WebAssembly runtime to marshal `Span` and `ArraySegment`.
+  * Unlike marshaling an array, marshaling a `Span` or `ArraySegment` doesn't create a copy of the underlying memory.
+  * `MemoryView` can only be properly instantiated by the .NET WebAssembly runtime. Therefore, it isn't possible to import a JS method as a .NET method that has a parameter of `Span` or `ArraySegment`.
+  * `MemoryView` created for a `Span` is only valid for the duration of the interop call. As `Span` is allocated on the call stack, which doesn't persist after the interop call, it isn't possible to export a .NET method that returns a `Span`.
+  * `MemoryView` created for an `ArraySegment` survives after the interop call and is useful for sharing a buffer. Calling `dispose()` on a `MemoryView` created for an `ArraySegment` disposes the proxy and unpins the underlying .NET array. We recommend calling `dispose()` in a `try-finally` block for `MemoryView`.
+
+Some combinations of type mappings that require nested generic types in [`JSMarshalAs`](xref:System.Runtime.InteropServices.JavaScript.JSMarshalAsAttribute%601) aren't currently supported. For example, attempting to materialize an array from a [`Promise`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise) such as `[return: JSMarshalAs<JSType.Promise<JSType.Array<JSType.Number>>>()]` generates a compile-time error. An appropriate workaround varies depending on the scenario, but this specific scenario is further explored in the [Type mapping limitations](#type-mapping-limitations) section.
+
+## JS primitives
+
+The following example demonstrates `[JSImport]` leveraging type mappings of several primitive JS types and the use of [`JSMarshalAs`](xref:System.Runtime.InteropServices.JavaScript.JSMarshalAsAttribute%601), where explicit mappings are required at compile time.
+
+`PrimitivesShim.js`:
+
+```javascript
+globalThis.counter = 0;
+
+// Takes no parameters and returns nothing.
+PrimitivesShim.IncrementCounter = function () {
```

Comment:
the syntax is typically in JS with lowerCase first letter
`export function incrementCounter() {`

which means you need to also drop `PrimitivesShim` and lowerCase it in JSImport
[JSImport("incrementCounter", "PrimitivesShim")]

Please test it all after you are done with updates.

Reply:
@SerratedSharp ... I'll make the updates, but can you test? I'm completely buried in other high priority work ⛰️⛏️😅. I didn't test any of the code for the new article. If testing has to wait for me to get free, this PR will be held up for a while.